hadrons/utils/track_distribution.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+from numpy.random import Generator, default_rng
+
+
+def create_track_distribution(
+    computation_time_steps: int,
+    dt: float,
+    number_of_tracks: int,
+    separation_time_steps: int,
+    random_generator: Generator = default_rng(2137),
+):
+    # Time when the tracks can be generated
+    beam_generation_time = (computation_time_steps - separation_time_steps) * dt
+
+    # Generate a random delay for each beam (between 0, 1)
+    track_delay = random_generator.random(number_of_tracks)
+
+    # Make the delays relative to each other (add the sum of previous delays to each element)
+    summed_delays = np.cumsum(track_delay)
+
+    # Normalize the delays to fit in the beam generation time
+
+    # two implementations: 1st is the original, it's biased towards the end of the beam generation
+    # time since it normalizes via the last element, the 2nd implementation normalizes by the
+    # number of beams - this ensures a uniform distribution of beams over the beam generation time
+    # TODO: check if the 1st implementation bias was intentional or accidental
+
+    # normalized_delays = summed_delays / summed_delays[-1] * beam_generation_time
```

Comment:
@jbrage I've noticed that the beam distribution was biased towards the later bins since we normalize the times by the last value (so the last track would always be put in the last bin - it would be simulated right before the simulation time steps). The new code below should simulate an even distribution across all time steps (before separation).

This new even distribution can sometimes fall in the first few time steps and get discarded by the code I mentioned in the other comment. This results in the simulation not calculating the recombination correctly (since we didn't count any initialized charge carriers).

Should we bias the distribution towards the later time steps?

Reply:
@retkiewi you are right: the distribution should not be biased towards the later time steps, that is a mistake. The distribution should be even across all bins